### PR TITLE
Add content/en/technical-debt

### PR DESCRIPTION
**Pull request by gullapalli9482**
Any feedback is highly appreciated.

### CODE REVIEW

**Comment by seokho-son** [comment]
ref #452

**Comment by seokho-son** [comment]
@gullapalli9482 Thank you for the contribution ! :)

I think this PR is not complete yet.
Please check the style-guide https://glossary.cncf.io/style-guide/#three-subheadings

**Comment by iamNoah1** [comment]
This is a hard one, I worried that this would happen. Technical debt somehow is a concept, but there is no problem it solves ^^. If I were to work on that one I would probably have chosen the "property" category. What do you think @JasonMorgan?

**Comment by gullapalli9482** [comment]
Just like @iamNoah1 said,  when I was working on this, it felt more like property, than a concept since it wasn't addressing any particular problem.
I'll create a separate pr by adding the property tag.
Do let me know about the content written, in case there is some addition needed to be done to it.

**Review by CathPag — COMMENTED** [review]
Thanks, @gullapalli9482. It's a good start but I think this definition needs some work. 

The definition is hard to follow and I had to read it a few times to understand what you meant. Here are two great resources that may help:

In this post by [Martin Fowler](https://martinfowler.com/bliki/TechnicalDebt.html), it's clear that tech debt is unavoidable and part of the software development process. In your definition it sound like only organizations that do shortcuts, accumulate tech debt.

[This Red Hat](https://enterprisersproject.com/article/2020/6/technical-debt-explained-plain-english) article is also great.  It explains the concept in easy terms and might serve as inspiration to achieve the simplicity we are looking for. 

Also, please make sure to submit a clean text. Punctuation is all over the place and there are articles missing — all things a regular spellcheck should flag. We highly recommend starting with a Google or Word doc as it will flag all these things. We are happy to help polishing the definition but everything a spellcheck can catch, should be addressed before submitting.

Let me know if you have any other questions.

**Comment by gullapalli9482** [comment]
I'll make sure it's better the next time.

**Comment by CathPag** [comment]
@gullapalli9482, just checking if you're still working on this.

**Comment by gullapalli9482** [comment]
Yes , yes I am

**Comment by gullapalli9482** [comment]
I'm sorry about the number of blunders I'm making and not doing proper homework before submitting my work.

File: content/en/technical-debt.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: Technical Debt
+status: Feedback Appreciated
+category: property
+---
+
+
+Technical debt is a concept in software development that reflects the implied cost of additional rework caused by choosing an easy solution for now instead of using a better approach that would take longer. It is a part of every software development project since trade-offs between speed, cost, and quality are impossible to avoid. Technical debts could arrive due to various defects in documentation, code, automation, design, architecture, etc. The best practices to counter technical debt include
```
Comment (CathPag) [inline content/en/technical-debt.md:8]:
```suggestion
Technical debt is a concept in software development that reflects the implied cost of additional rework. Applications are continuously updated and improved, and today's technical decisions will impact tomorrow's changes. Caused by trade-offs between speed, cost, and quality, technical debt is impossible to avoid completely. Technical debts can result from errors in documentation, code, automation, design, architecture, etc. 
```

Comment (CathPag) [reply content/en/technical-debt.md:8]: Added a sentence as to why change is needed and removed "caused by choosing an easy over a better approach" as it sounds too judgmental. The rest are just stylistic changes.

File: content/en/technical-debt.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+1) Establishing proper architecture. Designing a fitting architecture for the project avoids the majority of bugs.
```
Comment (CathPag) [inline content/en/technical-debt.md:9]: Not sure we should add best practices here. Explaining what it is, should be enough for a glossary definition. I'd skip this section completely.

**Review by CathPag — COMMENTED** [review]
Looks great, @gullapalli9482! Made a suggestion for the first part and a comment for the second.

**Comment by gullapalli9482** [comment]
I've committed the suggested changes @CathPag.

File: content/en/technical-debt.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+title: Technical Debt
+status: Feedback Appreciated
+category: property
+---
+
+
+Technical debt is a concept in software development that reflects the implied cost of additional rework. Applications are continuously updated and improved, and today's technical decisions will impact tomorrow's changes. Caused by trade-offs between speed, cost, and quality, technical debt is impossible to avoid completely and can result from errors in documentation, code, automation, design, architecture, etc.
```
Comment (CathPag) [inline content/en/technical-debt.md:8]:
```suggestion
Technical debt is a concept in software development that reflects the implied cost of additional rework. It's a metaphor that helps frame this rework like financial debt — the additional effort required to add new features represents the interest paid on the debt.

Applications are continuously updated and improved, and today's technical decisions will impact tomorrow's changes. Caused by trade-offs between speed, cost, and quality, technical debt is impossible to avoid completely and can result from errors in documentation, code, automation, design, architecture, among others. 
```

**Review by CathPag — COMMENTED** [review]
I think we should add the metaphor explanation as it really helps understand the concept.

**Comment by gullapalli9482** [comment]
Agreed, committed the suggested changes.

**Comment by gullapalli9482** [comment]
Are any changes yet to be made?

**Comment by CathPag** [comment]
> Are any changes yet to be made?

Yes, please remove the text after line 11.

**Comment by gullapalli9482** [comment]
Done 👍@CathPag

**Comment by gullapalli9482** [comment]
I feel that's pretty much the changes that were to be made.

**Review by iamNoah1 — APPROVED** [review]
@gullapalli9482 Thank you for the contribution :) The content looks good IMHO. At the moment we are in the process of figuring out where to draw the line between terms that are suitable for the Cloud Native Glossary and which of them are not. The request for this term was done before we had the triage process in place, where we accept or decline terms. So we will need you to be a little bit patient until we decided if this term should make it to the glossary or not :)

**Comment by gullapalli9482** [comment]
😀 cool

**Comment by CathPag** [comment]
@gullapalli9482, first of all, I wanted to say how grateful we are for your work. Unfortunately, we've decided that technical debt isn't a term that should go into the glossary. After consulting with the CNCF, we decided that terms must follow the [CNCF's cloud native definition ](https://github.com/cncf/toc/blob/main/DEFINITION.md) and technical debt does not fit that requirement. Thanks for your efforts and I hope we can still get you to contribute to the glossary going forward.

Thanks, and our sincere apologies!
